plugins/modules/ec2_ami.py

```diff
@@ -484,19 +497,23 @@ def create_image(module, connection):
                 params['KernelId'] = kernel_id
             if root_device_name:
                 params['RootDeviceName'] = root_device_name
-            image_id = connection.register_image(**params).get('ImageId')
+            image_id = connection.register_image(aws_retry=True, **params).get('ImageId')
     except (botocore.exceptions.BotoCoreError, botocore.exceptions.ClientError) as e:
         module.fail_json_aws(e, msg="Error registering image")
 
     if wait:
         waiter = connection.get_waiter('image_available')
         delay = wait_timeout // 30
         max_attempts = 30
-        waiter.wait(ImageIds=[image_id], WaiterConfig=dict(Delay=delay, MaxAttempts=max_attempts))
+        try:
+            # This isn't perfect: throttling errors mid-wait would reset the count.
+            AWSRetry.jittered_backoff()(waiter.wait)(ImageIds=[image_id], WaiterConfig=dict(Delay=delay, MaxAttempts=max_attempts))
+        except (botocore.exceptions.BotoCoreError, botocore.exceptions.ClientError, botocore.exceptions.WaiterError) as e:
```

Comment:
WaiterError is handled by BotoCoreError
```suggestion
        except (botocore.exceptions.BotoCoreError, botocore.exceptions.ClientError) as e:
```